synfig-core/src/modules/mod_webp/mptr_webp.cpp

```diff
@@ -0,0 +1,198 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file mptr_webp.cpp
+**	\brief WEBP (via FFMPEG) Importer Module
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**	Copyright (c) 2007 Chris Moore
+ **	Copyright (c) 2022 BobSynfig
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+**
+** ========================================================================= */
+
+/* === H E A D E R S ======================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include "mptr_webp.h"
+
+#include <ETL/stringf>
+
+#include <synfig/general.h>
+#include <synfig/localization.h>
+
+#if HAVE_IO_H
+ #include <io.h>
+#endif
+#if HAVE_PROCESS_H
+ #include <process.h>
+#endif
+#if HAVE_FCNTL_H
+ #include <fcntl.h>
+#endif
+#include <iostream>
+#endif
+
+/* === M A C R O S ========================================================= */
+
+using namespace synfig;
+using namespace etl;
+
+/* === G L O B A L S ======================================================= */
+
+SYNFIG_IMPORTER_INIT       ( webp_mptr );
+SYNFIG_IMPORTER_SET_NAME   ( webp_mptr, "webp" );
+SYNFIG_IMPORTER_SET_EXT    ( webp_mptr, "webp" ); // not working look at the main.cpp for list opf available extensions
+SYNFIG_IMPORTER_SET_VERSION( webp_mptr, "0.1"  );
+SYNFIG_IMPORTER_SET_SUPPORTS_FILE_SYSTEM_WRAPPER(webp_mptr, false);
+
+/* === M E T H O D S ======================================================= */
+
+bool webp_mptr::is_animated()
+{
+	return true;
+}
+
+bool
+webp_mptr::seek_to(const Time& time)
+{
+	//if(frame<cur_frame || !file)
+	//{
+		pipe = nullptr;
+
+		// FIXME: 24 fps is hardcoded now, but in fact we have to get it from canvas
+		//float position = (frame+1)/24; // webp didn't work with 0 frame
+		//float position = 1000/24; // webp didn't work with 0 frame
+		const std::string position = time.get_string(Time::FORMAT_NORMAL);
+
+		OS::RunArgs args;
+		args.push_back( { "-ss",      position                              });
+		args.push_back( {"-i",        filesystem::Path(identifier.filename) });
```

Comment:
I made a separated module to avoid to introduce complexity and separation of concerns.
Based on ffmpeg module, it has different parameters, also a different configuration dialog.
"Multiplexed code" is adding conditions and removing lisibility (intention -> goal -> result).
That's also the reason I like vertical alignment, then source has to be seen in its globality, not line-by-line.
And a separated module also permits to avoid to mess the original code with such formatting.